Paws4Me/Paws4Me/Display And Save Pets/View Models/petLocaldatabaseViewModel.swift

```diff
@@ -0,0 +1,117 @@
+//
+//  petLocaldatabaseViewModel.swift
+//  Paws4Me
+//
+//  Created by Kirsty-Lee Walker on 2022/03/25.
+//
+
+import Foundation
+
+// MARK: - PetLocalDatabaseViewModel Delegate
+protocol PetLocalDatabaseViewModelDelegate: AnyObject {
+    func reloadView()
+    func showError(errorTitle: String, errorMessage: String, action: LocalDatabaseError)
+}
+
+class PetLocaldatabaseViewModel {
+
+    // MARK: - Vars/Lets
+    private var petLocalDatabaseRepository: SavedPetDataRepository?
+    private weak var delegate: PetLocalDatabaseViewModelDelegate?
+    private var pets: [Pet]? = []
+    private var namePet = ""
+    private var imagePet = ""
+    var isDeleteSucess = false
+
+    // MARK: - Constructor
+    init(repository: SavedPetDataRepository,
+         delegate: PetLocalDatabaseViewModelDelegate) {
+        self.petLocalDatabaseRepository = repository
+        self.delegate = delegate
+    }
+
+    // MARK: - Functions
+    func setNamePet(name: String) {
```

Comment:
Naming enhancement:

`func set(petName: String)`

Please apply to other functions as well 😄

Reply:
This is.a much better naming convention. Thank you 👍🏻